database/rrd.h

```diff
@@ -919,6 +928,14 @@ extern RRDSET *rrdset_find(RRDHOST *host, const char *id);
 
 extern RRDSET *rrdset_find_bytype(RRDHOST *host, const char *type, const char *id);
 #define rrdset_find_bytype_localhost(type, id) rrdset_find_bytype(localhost, type, id)
+/* This will not return charts that are archived */
```

Comment:
The original `rrdset_find_bytype` in `rrset.c` should be removed instead of left as dead code.

I there a reason to move this to a `static inline` definition in a header file instead of an `inline` definition in the `.c` file like the original?

Reply:
`rrdset_find_bytype()` is not dead code, it's being called by `rrdset_find_bytype_localhost()`, which is called by `rrdset_find_active_bytype_localhost()`. You could argue that `rrdset_find_bytype_localhost()` is dead code though.

When writing `rrdset_find_active_bytype_localhost()` I just wanted to make as few changes as possible, so I didn't refactor anything around it.